libc/math/trig.c

```diff
@@ -0,0 +1,697 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ *
+ * HEADER:
+ *    - math.h
+ *
+ * TESTED:
+ *    - sin(), cos(), tan()
+ *    - asin(), acos(), atan(), atan2()
+ *    - sinh(), cosh(), tanh()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <math.h>
+#include <limits.h>
+#include <float.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <time.h>
+
+#include "common.h"
+#include <unity_fixture.h>
+
+
+TEST_GROUP(math_trig);
+TEST_GROUP(math_hyper);
+
+
+TEST_SETUP(math_trig)
+{
+}
+
+
+TEST_TEAR_DOWN(math_trig)
+{
+}
+
+
+TEST(math_trig, sin_basic)
+{
+	int i, iters = 50 * ITER_FACTOR;
+	double acceptLoss = 35.0;
+	double max = M_PI_2;
+	double min = -max;
+	double step = max / iters - min / iters;
+	double xstart = min;
+	double u, x, y, f, g, digLost;
+
+	for (i = 0; i < iters; i++) {
+		u = (double)rand() / (double)RAND_MAX;
+		x = step * u + xstart;
+		/* Using sinus trigonometric identity */
```

Comment:
What identity exactly? Would be nice to add e.g. equations used in all such instances for a future reference. It's not 100% obvious from the code alone (i.e. requires some analysis from the reader to find out identity used).

Reply:
These identities do not have particular names and I think they are pretty self-explanatory, in most cases `g` is given function and `f` represents identity, which has to be equal.

Reply:
There are many places where code looks similar but there are often few differences, what would be hard to abstract it under single function with many arguments and it would cause readability decrease IMHO.